packages/calling/src/CallingClient/calling/call.ts

```diff
@@ -2794,6 +2810,55 @@ export class Call extends Eventing<CallEventTypes> implements ICall {
   getCallRtpStats(): Promise<CallRtpStats> {
     return this.getCallStats();
   }
+
+  /**
+   * Create a timeout for the next expected state
+   * @param expectedStates - An array of next expected states
+   * @param timeoutDuration - Timeout duration in milliseconds after which the call will be disconnected
+   * @param errorMessage - Error message to be emitted if the call is not in the expected state in expected time
+   */
+  private setTimeoutForState(timeoutDuration: number) {
+    this.callEventsTimeout = setTimeout(() => {
+      const expectedStates = this.callStateMachine.state.nextEvents;
+      const currentState = this.callStateMachine.state.value;
+      this.handleTimeout(expectedStates, `Call timed out in state: ${currentState}`);
+    }, timeoutDuration);
+  }
+
+  /**
+   * Handle timeout for the missed events
+   * @param expectedStates - An array of next expected states
+   * @param errorMessage - Error message to be emitted if the call is not in the expected state in expected time
+   */
+  private async handleTimeout(expectedStates: string[], errorMessage: string) {
+    if (
+      this.callStateMachine.state.value &&
+      !expectedStates.includes(this.callStateMachine.state.value.toString())
+    ) {
+      log.warn(errorMessage, {
+        file: CALL_FILE,
+        method: 'handleTimeout',
+      });
+      this.deleteCb(this.getCorrelationId());
+      this.emit(CALL_EVENT_KEYS.DISCONNECT, this.getCorrelationId());
+      const response = await this.delete();
+
+      log.log(`handleTimeout: Response code: ${response.statusCode}`, {
```

Comment:
Should this be log.log or log.info?

Reply:
It should be log.log only. We are just logging the response of the delete method. This is similar to how we are doing in other methods